pkg/cmd/helm/build/build.go

```diff
@@ -154,21 +168,64 @@ func (o *Options) Run() error {
 		if err != nil {
 			return errors.Wrapf(err, "failed to lint")
 		}
+		if o.OCI {
+
+			if o.RepositoryPassword == "" {
+				log.Logger().Debugf("OCI helm dependency build using --registry-config  %s", info(o.RegistryConfigFile))
+				c = &cmdrunner.Command{
+					Dir:  chartDir,
+					Name: o.HelmBinary,
+					Args: []string{"dependency", "build", ".", "--registry-config", o.RegistryConfigFile},
+				}
+				_, err = o.CommandRunner(c)
+				if err != nil {
+					return errors.Wrapf(err, "failed to build dependencies")
+				}
+			} else {
+				log.Logger().Debugf("OCI helm dependency build using username/password '%s/***'", info(o.RepositoryUsername))
 
-		c = &cmdrunner.Command{
-			Dir:  chartDir,
-			Name: o.HelmBinary,
-			Args: []string{"dependency", "build", "."},
-		}
-		_, err = o.CommandRunner(c)
-		if err != nil {
-			return errors.Wrapf(err, "failed to build dependencies")
+				c = &cmdrunner.Command{
+					Dir:  chartDir,
+					Name: o.HelmBinary,
+					Args: []string{"registry", "login", o.RepositoryURL, "--username", o.RepositoryUsername, "--password", o.RepositoryPassword},
+				}
+				_, err = o.CommandRunner(c)
+				if err != nil {
+					return errors.Wrapf(err, "failed to helm login")
+				}
+				c = &cmdrunner.Command{
+					Dir:  chartDir,
+					Name: o.HelmBinary,
+					Args: []string{"dependency", "build", "."},
+				}
+				_, err = o.CommandRunner(c)
+				if err != nil {
+					return errors.Wrapf(err, "failed to build dependencies")
+				}
+			}
+		} else {
+			c = &cmdrunner.Command{
+				Dir:  chartDir,
+				Name: o.HelmBinary,
+				Args: []string{"dependency", "build", "."},
+			}
+			_, err = o.CommandRunner(c)
+			if err != nil {
+				return errors.Wrapf(err, "failed to build dependencies")
+			}
 		}
-
-		c = &cmdrunner.Command{
-			Dir:  chartDir,
-			Name: o.HelmBinary,
-			Args: []string{"package", "."},
+		if o.OCI && o.RepositoryPassword == "" {
+			c = &cmdrunner.Command{
+				Dir:  chartDir,
+				Name: o.HelmBinary,
+				Args: []string{"package", ".", "--version", o.Version, "--registry-config", o.RegistryConfigFile},
+			}
+		} else {
+			c = &cmdrunner.Command{
+				Dir:  chartDir,
+				Name: o.HelmBinary,
+				Args: []string{"package", ".", "--version", o.Version},
```

Comment:
I don't know if --version is mandatory for the OCI case, but in any case I don't see why it should start to be mandatory for other cases.

Reply:
I will test it out. Should probably be removed